### ISSUE TITLE
loss function does not match in CNTK_303_Deep_Structured_Semantic_Modeling_with_LSTM_Networks

### ISSUE THREAD

**Issue by tfrichard** [Microsoft/CNTK#3182]
Hi CNTK experts,

I am developing a Dssm model by following the CNTK tutorial but ran into a weird problem on loss value in training. I find the loss value given by summarize_training_progress is different from average loss directly given by network loss function:

Finished Epoch[1 of 50]: [Training] loss = 0.245744 * 1024, metric = 9.47% * 1024 7.775s (131.7 samples/s);
Epoch: 1 Loss: 0.878567 Error: 0.570312

Finished Epoch[2 of 50]: [Training] loss = 0.115569 * 1024, metric = 3.52% * 1024 0.050s (20480.0 samples/s);
Epoch: 2 Loss: 0.856831 Error: 0.500000

Finished Epoch[3 of 50]: [Training] loss = 0.068359 * 1024, metric = 2.15% * 1024 0.047s (21787.2 samples/s);
Epoch: 3 Loss: 0.861055 Error: 0.484375

Finished Epoch[4 of 50]: [Training] loss = 0.047108 * 1024, metric = 0.98% * 1024 0.055s (18618.2 samples/s);
Epoch: 4 Loss: 0.888497 Error: 0.507812

Finished Epoch[5 of 50]: [Training] loss = 0.046957 * 1024, metric = 1.46% * 1024 0.050s (20480.0 samples/s);
Epoch: 5 Loss: 0.887553 Error: 0.460938

Finished Epoch[6 of 50]: [Training] loss = 0.040135 * 1024, metric = 1.07% * 1024 0.043s (23814.0 samples/s);
Epoch: 6 Loss: 0.901011 Error: 0.476562

Here’s my simple loss calculator:

def my_progress_report(network, data, epoch):
    loss = network['loss'].eval(data)
    avg_loss = np.average(loss)    
    error = network['error'].eval(data)
    avg_error = np.average(error)
print("Epoch: {0} Loss: {1:8f} Error: {2:8f}".format(epoch, avg_loss, avg_error))

This is given by CNTK and this is given by the loss function, both are feed into the same minibatch data. It’s obvious one is decreasing quickly but the other one does not.

Did anyone observe such behavior? I copied all the model creation codes and parameters from the tutorial. And even the tutorial itself has such problem.

Any help is high appreciated. Thanks in advance.

**Comment by tangyuq** [comment]
As we discussed offline, it is related to your dropout layers which behave differently during training and evaluation. After setting DROPOUT_RATIO = 0.0, the training and evaluation behavior become sync. I am closing this issue for now. Please feel free to reopen the issue if your have further concerns.

def create_model(q_input, d_input):
    with C.layers.default_options(initial_state=0.1):
        q_dssm = C.layers.Sequential([            
            C.layers.Dense(DSSM_DIM, activation=C.relu, name='q_l1'),
            C.layers.Dropout(DROPOUT_RATIO, name='dropout qdo1'),
            C.layers.Dense(DSSM_DIM, activation=C.relu, name='q_l2'),
            C.layers.Dropout(DROPOUT_RATIO, name='dropout qdo2'),
            C.layers.Dense(DSSM_DIM, activation=C.relu, name='q_l3')
        ])
        
        d_dssm = C.layers.Sequential([            
            C.layers.Dense(DSSM_DIM, activation=C.relu, name='d_l1'),
            C.layers.Dropout(DROPOUT_RATIO, name='dropout ddo1'),
            C.layers.Dense(DSSM_DIM, activation=C.relu, name='d_l2'),
            C.layers.Dropout(DROPOUT_RATIO, name='dropout ddo2'),
            C.layers.Dense(DSSM_DIM, activation=C.relu, name='d_l3')
        ])

    return  (q_dssm(q_input)), (d_dssm(d_input))
